admin/server/api/list/delete.js

```diff
@@ -17,13 +17,17 @@ module.exports = function (req, res) {
 	if (!Array.isArray(ids)) {
 		ids = [ids];
 	}
+
 	if (req.user) {
+		var checkResourceId = (keystone.get('user model') === req.list.key);
+
 		var userId = String(req.user.id);
-		if (ids.some(function (id) {
+		// check if user can delete this resources based on resources ids and userId
+		if (checkResourceId && ids.some(function (id) {
 			return id === userId;
 		})) {
-			console.log('Refusing to delete ' + req.list.key + ' items; ids contains current User');
-			return res.apiError(403, 'not allowed', 'You can not delete yourself');
+			console.log('Refusing to delete ' + req.list.key + ' items; resources ids contains current User id');
+			return res.apiError(403, 'not allowed', 'You can not delete this resource');
```

Comment:
Hi! This is because the main implementation only allows user do not delete yourself, but I'm thinking maybe this can be expanded sometimes to allow other kind of ACL, so I changed to a more generic message. (In the client application this is not the error message that appear for user)